### PR TITLE
Add Debug derive to std::process::Output

### PR DESCRIPTION
**Pull request by webmobster-zz**
I didn't see any reason that debug couldn't be added to this object, since every field derives debug.

### CODE REVIEW

**Comment by sfackler** [comment]
@bors r+

Thanks!

**Comment by sfackler** [comment]
Actually, I'm not sure the derived impl would be best here. Stdout and stderr are both `Vec<u8>`, so their output will be e.g. `[32, 20, 39, 20, ...]` which is not going to be great to look at.

**Comment by webmobster** [comment]
What would be the alternative assume the stdout/stder will always be utf8 or ascii and convert them outputs to a utf8 string?

**Comment by sfackler** [comment]
`String::from_utf8_lossy` is one option. It could maybe also see if it's UTF8 and if not fall back to the raw output, possibly truncated?

**Comment by brson** [comment]
Since stdout/err doesn't need to be text I prefer @sfackler's latter option, printing text if utf8 and binary if not. Truncated in each case I guess.

**Comment by webmobster** [comment]
If thats the consensus I can write up a debug implementation tomorrow (late here), should I close this pull request. I'll keep the pull request open till thats done I guess.

**Comment by sfackler** [comment]
Sounds good!

**Comment by webmobster** [comment]
What should I put as the stability annotiation on the function?

**Comment by sfackler** [comment]
The stability checking logic doesn't check trait implementations, so it's automatically stable.

**Comment by webmobster** [comment]
I got a "src/libstd/process.rs:406:1: 429:2 error: This node does not have a stability attribute",

**Comment by petrochenkov** [comment]
@webmobster 
`#[stable(feature = "process_output_debug", since = "1.7.0")]`
Even though these annotations are not checked now, they are still required for documentation and possible future checking.

File: src/libstd/process.rs [hunk 1]
```diff
@@ -400,6 +401,33 @@ pub struct Output {
     pub stderr: Vec<u8>,
 }
 
+//If either stderr or stdout are valid utf8 strings it prints the valid strings, otherwise it
+//prints the byte sequence instead
```
Comment (alexcrichton) [inline src/libstd/process.rs]: Stylistically there's typically a space after the `//`, and can you also format this comment to a maximum width of 80 chars? (the surrounding style)

File: src/libstd/process.rs [hunk 2]
```diff
@@ -400,6 +401,33 @@ pub struct Output {
...
+        let stdout_utf8 = str::from_utf8(&self.stdout);
+        let stdout_debug: &fmt::Debug = match stdout_utf8
+        {
+            Ok(ref str) => str,
+            Err(_) => &self.stdout
+        };
+
+        let stderr_utf8 = str::from_utf8(&self.stderr);
+        let stderr_debug: &fmt::Debug = match stderr_utf8
+        {
```
Comment (alexcrichton) [inline src/libstd/process.rs]: Stylistically the opening `{` typically comes on the same line as the `match`

**Comment by alexcrichton** [comment]
Looks good to me (including the stability annotation @petrochenkov mentioned)

**Comment by webmobster** [comment]
Done

**Comment by alexcrichton** [comment]
@bors: r+ 21030f1fc9510d9808c0f8352ea8a5eeacfa4ca5

Thanks!

**Pull request merged**